### DISCUSSION TITLE
Toga Window widget error on Android

### DISCUSSION THREAD

**Discussion by wanglifeng95** [beeware/toga#2016] [Q&A] [answered]
At first, I thought the Toga Window widget is an Activity on Android, which is a window that renders the UI. After closing, it returned to the previous window. But on Android, using the Toga Window widget can result in the following errors: 
AttributeError: 'Window' object has no attribute 'viewport'

Why is this? The code is as follows:
```
class WindowDemoApp(toga.App):

    def do_new_windows(self, widget, **kwargs):
        newWindow = toga.Window("Non-resizable Window",on_close=self.close_handler)
        newWindow.content = toga.Box(children=[toga.Label("It's a new window")])
        self.app.windows += newWindow
        newWindow.show()

    def close_handler(self, window, **kwargs):
        window.close()

    def startup(self):
        self.main_window = toga.MainWindow(title=self.name)
        self.label = toga.Label("Ready.")
        btn_do_new_windows = toga.Button(
            "Create Window", on_press=self.do_new_windows
        )
        self.main_box = toga.Box(
            children=[self.label,btn_do_new_windows],style=Pack(direction=COLUMN))

        self.main_window.content = self.main_box
        self.main_window.show()

def main():
    return WindowDemoApp()
```
Why did this error occur? If the Window widget cannot be used on Android, how can Toga implement the Android Activity function？

**Comment by freakboy3742** [comment]
Toga doesn't support additional windows on mobile platforms, because it's not clear what a "Window" would map to. On a desktop, you can have multiple windows visible at any given time, and can add and remove windows arbitrarily; on mobile, you can generally only see one window at a time. On Android specifically, an "Activity" has an entire lifecycle, making it a closer match for Application, not Window.

At present, there isn't any support for Android's concept of Activity, beyond the MainActivity. If you want to build your own supplemental activity, you'll need to build it using Chaquopy primitives.

**Reply by mhsmith** [reply, marked as answer]
If you want your app to contain multiple screens, currently the easiest way is to change the `main_window.content` attribute, which you can do as many times as you like.

**Reply by wanglifeng95** [reply]
Thanks very much. It's a good idea
